modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/trigger/messaging/ObjectActionsTriggerMessageListener.java

```diff
@@ -134,6 +151,32 @@ protected void doReceive(Message message) throws Exception {
 				objectDefinition, objectEntry, null,
 				_userLocalService.getUser(message.getLong("userId"))),
 			message.getLong("userId"));
+
+		if (!objectDefinition.isRootDescendantNode() &&
+			(!objectDefinition.isRootNode() ||
+			 StringUtil.equals(
+				 actionTrigger,
+				 ObjectActionTriggerConstants.KEY_ON_AFTER_ADD))) {
+
+			return;
+		}
+
+		ObjectEntry rootObjectEntry = _objectEntryLocalService.fetchObjectEntry(
+			objectEntry.getRootObjectEntryId());
+
+		if (rootObjectEntry == null) {
+			return;
+		}
+
+		_objectActionEngine.executeObjectActions(
+			objectDefinition.getClassName(), message.getLong("companyId"),
+			ObjectActionTriggerConstants.KEY_ON_AFTER_ROOT_UPDATE,
+			() -> ObjectEntryUtil.getPayloadJSONObject(
+				_dtoConverterRegistry, _jsonFactory,
+				ObjectActionTriggerConstants.KEY_ON_AFTER_ROOT_UPDATE,
+				objectDefinition, rootObjectEntry, null,
+				_userLocalService.getUser(message.getLong("userId"))),
+			message.getLong("userId"));
```

Comment:
() -> ObjectEntryUtil.getPayloadJSONObject(
	_dtoConverterRegistry, _jsonFactory,
	ObjectActionTriggerConstants.KEY_ON_AFTER_ROOT_UPDATE,
	**objectDefinition**, rootObjectEntry, null,
	_userLocalService.getUser(message.getLong("userId")))

- In ObjectEntryModelListener.java, the method does a GET in the objectEntry.
- In your case, the objectEntry is rootObjectEntry, and you did not do the GET. I think you must do it because the rootObjectEntry can belong to another ObjectDefinition.